tests/api/api/test_functions.py

```diff
@@ -1000,19 +1002,43 @@ def failing_func():
         assert background_task.status.state == expected_status_result
 
 
+def test_list_functions_by_foo_spec(
```

Comment:
add some more cases like creating a function that should be filtered out and make sure you don't get it in the response